File: 377-combination-sum-iv/377-combination-sum-iv.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        dp = [0] * (target + 1)
        dp[0] = 1
        for remaining in range(target + 1):
            for num in nums:
                if remaining - num < 0:
                    continue
                dp[remaining] += dp[remaining - num]
        return dp[target]

File: 377-combination-sum-iv/377-combination-sum-iv.java
class Solution {
    public int combinationSum4(int[] nums, int target) {
        int[] dp = new int[target + 1];
        dp[0] = 1;
        for(int remaining = 0; remaining <= target; remaining++)
            for(int num : nums){
                if(remaining < num)
                    continue;
                dp[remaining] += dp[remaining - num];
            }
        return dp[target];
    }
}